tianshou/policy/multiagent/mapolicy.py

```diff
@@ -36,7 +36,9 @@ def process_fn(self, batch: Batch, buffer: ReplayBuffer,
         # reward can be empty Batch (after initial reset) or nparray.
         has_rew = isinstance(buffer.rew, np.ndarray)
         if has_rew:  # save the original reward in save_rew
-            save_rew, buffer.rew = buffer.rew, Batch()
+            # Since we do not override buffer.__setattr__, here we use _meta to
+            # change buffer.rew, otherwise buffer.rew = Batch() has no effect.
+            save_rew, buffer._meta.rew = buffer.rew, Batch()
```

Comment:
It would be nice to find a way to avoid such trick in future release because it is error prone.